be/src/exprs/json_functions.cpp

```diff
@@ -157,11 +157,34 @@ Status JsonFunctions::extract_from_object(simdjson::ondemand::object& obj, const
         // we have to do some special treatment for the second elem of json path.
         // If the key is not found in json object, simdjson::NO_SUCH_FIELD would be returned.
         if (i == 1) {
-            HANDLE_SIMDJSON_ERROR(obj.find_field_unordered(col).get(tvalue),
-                                  fmt::format("unable to find field: {}", col));
+            if (col == "*") {
+                // There should be no jsonpath for this pattern, $.*
```

Comment:
`$.*` is a valid jsonpath.

Reply:
> `$.*` is a valid jsonpath.

For avro union type, there is always a  field in root object. just like this.
{"field_name": null}
{"field_name": {"field_type": data}}